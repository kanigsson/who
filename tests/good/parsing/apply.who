logic = ['a||] : 'a -> 'a -> prop
logic /\ : prop -> prop -> prop
logic !! ['a|r|'e] : ref(r,'a) -> <r 'e > -> 'a
parameter ! ['a|r|] (x : ref(r,'a)) : 'a, {r} =
  {}
  {r : !!x = r /\ !!x|old = !!x}
let apply ['a 'b ||'e] (f : 'a ->{'e} 'b) (x : 'a) =
  {}
  f x
  {}

let h [|r|] (x : ref(r,int)) =
  {}
  apply [{r}] (fun (z : int) -> {} !x {}) 5
  {}
