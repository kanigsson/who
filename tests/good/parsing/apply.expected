let apply ['a 'b||e] = (位(f:('a ->{|e } 'b)) -> (位(x:'a) -> (f x))) in
let h [|r|] = (位(x:ref(r,nat)) -> ((apply (位(z:nat) -> (! x))) 5)) in ()
